### DISCUSSION TITLE
What is the best way to extend Mastodon core?

### DISCUSSION THREAD

**Discussion by chualuoi** [mastodon/mastodon#22482] [Q&A]
Hi folks,

I'm looking into Mastodon document but there is no section addressing the above question yet. What is the best way to add new features and making changes to the database structure without messing up with Mastodon core? Does Mastodon support plugins?

Thanks!

**Comment by trwnh** [comment]
i think you would have to fork and apply your own patches on top of that. there is an open request for plugins but i'm not sure it's very feasible: #7717
